RoslynPluginGenerator/NuGet/NuGetPackageHandler.cs

```diff
@@ -178,5 +191,36 @@ private IPackage SelectLatestVersion(IList<IPackage> packages)
 
             return package;
         }
+
+        private void GetAllDependencies(IPackage current, List<IPackage> collectedDependencies)
+        {
+            this.logger.LogDebug(UIResources.NG_ResolvingPackageDependencies, current.Id, current.Version);
+            Debug.Assert(current != null);
+
+            foreach (PackageDependency dependency in current.GetCompatiblePackageDependencies(null))
+            {
+                IPackage dependencyPackage = this.packageManager.LocalRepository.ResolveDependency(dependency, true, true);
```

Comment:
Rather than walking all of the dependency sets, we're now asking the local repository to resolve the required dependencies. It should be able to do this, because we're assuming that "FetchAssemblies" has been called first which will ensure that the primary package and the dependencies are installed.